ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -980,10 +980,16 @@ class JITModBuilder {
 
         if (isa<CallInst>(I)) {
           if (isa<IntrinsicInst>(I)) {
-            // `llvm.lifetime.start.*` and `llvm.lifetime.end.*` are not really
-            // calls, but rather left-in annotations used in LLVM optimisation
-            // passes. Since these "calls" generate no machine code, nor have
-            // program semantics, we can simply ignore them.
+            // `llvm.lifetime.start.*` and `llvm.lifetime.end.*` are
+            // annotations added by some compiler front-ends to allow backends
+            // to perform stack slot optimisations.
+            //
+            // They are not really calls, and since they generate no machine
+            // code, nor have program semantics, we can simply ignore them.
```

Comment:
On reflection I'm not keen on "nor have program semantics" which is easily misread. How about something like "Removing these annotations makes generated code slightly less efficient, but does not affect correctness, so we remove them to make our lives easier." ?

Reply:
Fixed in [5e85507](https://github.com/ykjit/yk/pull/528/commits/5e8550799cdb350b7cf62adca9f0e89cac3a6012)